submit-api/src/submit_api/models/queries/project.py

```diff
@@ -47,15 +47,30 @@ def filter_by_search_criteria(cls, query, search_options):
         if not search_options:
             return query
 
-        query = cls._filter_by_search_text(query, search_options.search_text)
+        query = cls._filter_by_submission_name(
+            query, search_options.search_text)
+
+        query = cls._filter_by_submission_status(query, search_options.status)
 
         return query
 
     @classmethod
-    def _filter_by_search_text(cls, query, search_text):
-        """Filter by search text across project name and package name."""
+    def _filter_by_submission_name(cls, query, search_text):
+        """Filter by search text across package name."""
         if search_text:
             query = query.join(Package, AccountProject.packages).filter(
                 Package.name.ilike(f"%{search_text}%")
             )
         return query
+
+    @classmethod
+    def _filter_by_submission_status(cls, query, statuses):
+        """Filter by submission status using overlap."""
+        if statuses:
+            # Convert enum to string values
+            status_values = [status.value for status in statuses]
+            if status_values:
+                query = query.join(Package, AccountProject.packages).filter(
```

Comment:
let's not join again on each filter, join is an expensive operation. in the main function join with Account project there and in the filter functions just do filter